packages/dds/migration-shim/src/shimDeltaConnection.ts

```diff
@@ -45,11 +51,61 @@ export class ShimDeltaConnection implements IDeltaConnection {
 		// preventing them from being processed by the tree delta handler, and the tree delta handler to process tree
 		// ops. Post migration v1 ops can be considered "shim" ops as they are dropped.
 		this.shimDeltaHandler.attachTreeDeltaHandler(handler);
-		if (!this.isShimDeltaHandlerAttachedToConnection) {
+		if (!this.shimDeltaHandler.attached) {
+			this.deltaConnection.attach(this.shimDeltaHandler);
+			this.shimDeltaHandler.markAttached();
+		}
+	}
+	public dirty(): void {
+		this.deltaConnection.dirty();
+	}
+
+	// This needs to be more thoroughly thought through. What happens when the source handle is changed?
+	public addedGCOutboundReference?(srcHandle: IFluidHandle, outboundHandle: IFluidHandle): void {
+		this.deltaConnection.addedGCOutboundReference?.(srcHandle, outboundHandle);
+	}
+}
+
+/**
+ * A delta connection that stamps ops with a particular channel attributes so that those ops won't get dropped
+ */
+export class StampDeltaConnection implements IDeltaConnection {
```

Comment:
Does this need to be its own whole class?  Vs. its responsibilities living somewhere else.

Reply:
I think so, because the LegacySharedTree points to the deltaConnection, maybe we could have one generic class with the ability to stamp the v2 op, but I don't know if that's the best way to do it. Let's merge these changes, and refactor them. I agree, this is ugly.